docs/source/format/Changing.rst

```diff
@@ -0,0 +1,67 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+How to Change Format Specification
+==================================
+
+**Cross-language** is important in Apache Arrow. To keep it, we use
+the following process when dealing with format (files in
+`<https://github.com/apache/arrow/tree/main/format>`_) changes:
+
+* We must go through a formal DISCUSS and VOTE process
+* We must have at least 2 reference implementations and associated
+  integration tests
+
+DISCUSS and VOTE Process
+------------------------
+
+DISCUSS process is a process to discuss the format changes in
+public. Anyone can join the discussion. The DISCUSS process is done by
+starting a thread in dev@arrow.apache.org with ``[DISCUSS]`` prefixed
+subject.
+
+.. note::
+
+   We sometimes use ``[Discuss]``, ``DISCUSS: `` or something but
+   ``[DISCUSS]`` is recommended.
+
+Here are examples:
+
+* `[Discuss][Format] Add 32-bit and 64-bit Decimals <https://lists.apache.org/thread/9ynjmjlxm44j2pt443mcr2hmdl7m43yz>`_
+* `[DISCUSS][Format] Starting to do some concrete work on the new "StringView" columnar data type <https://lists.apache.org/thread/dccj1qrozo88qsxx133kcy308qwfwpfm>`_
+
+VOTE process is a process to tell whether we have reached
+consensus. We can start a vote for the format changes after we reach
+consensus in the preceding DISCUSS process.
+
+See also: `Apache Voting Process <https://www.apache.org/foundation/voting.html>`_
+
+At Least 2 Reference Implementations
+------------------------------------
+
+We must have at least 2 reference implementations and associated
+integration tests to confirm whether the format changes are
+cross-compatible and consistent.
+
+We must choose complete implementations for them. Here are candidate
```

Comment:
How do we define a reference implementation is complete? For example, if a new data/array type is proposed, does the IPC, compute functions, and parquet module should be implemented as well?

Reply:
I don't think we need to lay down hard rules. Below we link to https://arrow.apache.org/docs/status.html, which gives us a grid of implementations and their status. Which parts are relevant will depend on the feature IMO. For most new data types, I suspect what's important is IPC implementation.

Reply:
It's ~up to the voters, effectively. I think it'd be hard to pin this down in text.

That said, this part is discussing whether the implementation is considered a reference Arrow implementation. Which is more about whether the the implementation is self-contained (as opposed to a binding) and ~completeness of the Arrow specification and development activity.